### ISSUE TITLE
Reduce occurrences of "multiple items with the same description" in the Gallery

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#21647]
Per Android (and probably iOS) accessibility guidelines, we should not have multiple semantics objects on the same page with identical descriptions. We should make sure we have properly disambiguated all occurrences by providing additional context in the label or choosing new names.

Known instances:

- buttons_demo (RAISED BUTTON, DISABLED)
- cards_demo (SHARE, EXPLORE)
- chip_demo (Poker, Tortilla, Fish and, Micro, Wood, Hammer, ...)
- data_table_demo
   * This one is potentially out of scope for this bug, but each cell label should have row + column attached.
- date_and_time_picker: The current time will result in the same label on both dropdown buttons
- icons_demo: Size, Enabled, Disabled

**Comment by SUPERETDUPER** [comment]
So for example the icons_demo we should just remove the second example? Is that what you mean?

**Comment by jonahwilliams** [comment]
We definitely shouldn't solve this by removing demos. Instead we'll need to make sure all of the different elements have meaningful semantic labels. In a real app this would be more easily solved, but as a gallery we need to be a little creative.

For example, In the icons demo, each Icon could be labeled with it's state (Size/enabled)

**Comment by jonahwilliams** [comment]
This is for screen reader support

**Comment by SUPERETDUPER** [comment]
Aha ok. Thank you!

**Comment by SUPERETDUPER** [comment]
I notice there is a `Semantics` widget that can be wrapped around other widgets to add a "invisible" label to the object.

For example for the button demo : 


    ButtonBar(
            mainAxisSize: MainAxisSize.min,
            children: <Widget>[
              Semantics.fromProperties(
                  child: RaisedButton.icon(
                    icon: const Icon(Icons.add, size: 18.0),
                    label: const Text('RAISED BUTTON'),
                    onPressed: () {
                      // Perform some action
                    },
                  ),
                  container: true,
                  excludeSemantics: true,
                  properties: const SemanticsProperties(
                      button: true, label: 'Raised button with icon')),
              Semantics.fromProperties(
                  container: true,
                  excludeSemantics: true,
                  properties: const SemanticsProperties(
                      button: true, label: 'Disabled button with icon'),
                  child: RaisedButton.icon(
                    icon: const Icon(Icons.add, size: 18.0),
                    label: const Text('DISABLED'),
                    onPressed: null,
                  )),
            ],
          )

This fixes the screen reader. However, the tests still fail for some reason.

    ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
    The following TestFailure object was thrown running a test:
    Expected: Text contrast should follow WCAG guidelines
    Actual: <Instance of 'WidgetTester'>
    Which: Multiple nodes with the same label: RAISED BUTTON
               Multiple nodes with the same label: DISABLED


Is this the best way of fixing the issue and if so, why do the tests still fail? Is it a bug?

**Comment by SUPERETDUPER** [comment]
Also why is the contrast guideline checking for labels and Text To Speech? Shouldn't that be another separate guideline?

**Comment by SUPERETDUPER** [comment]
@jonahwilliams ok sounds good! I don't have a lot of time to work on this right now but when I do I'll come back to it. Thank you!

**Comment by eseidelGoogle** [comment]
This is over a year old, and the Gallery has been reworked again.  Not clear this is still useful to keep open.

Please re-open if you disagree @willlarche
